src/index.ml

```diff
@@ -648,10 +628,11 @@ struct
               Tbl.clear log.mem;
               hook `After_clear;
               let log_async = assert_and_get t.log_async in
+              let append_io = IO.append log.io in
               Tbl.iter
                 (fun key value ->
                   Tbl.replace log.mem key value;
-                  append_key_value log.io key value)
+                  Entry.encode' key value append_io)
```

Comment:
Can you always inline the `append_io`? (In the call to `encode`, the `IO.append` sometimes it's inlined and sometimes it's not.)

Reply:
Not inlining here saves allocating a closure for each entry in the table (and similarly for the `iter_io`), but I can still inline it if you prefer?

Reply:
Ah, I missed that. Keep it as is it then.